tests/test_efficientnet.py

```diff
@@ -0,0 +1,285 @@
+        device = "cuda" if torch.cuda.is_available() else "cpu"
+
+        # open image
+        image_size = get_efficientnet_image_size(input_param["model_name"])
+        img = PIL.Image.open(image_path)
+
+        # defin ImageNet transform
+        tfms = torchvision.transforms.Compose(
+            [
+                torchvision.transforms.Resize(image_size),
+                torchvision.transforms.CenterCrop(image_size),
+                torchvision.transforms.ToTensor(),
+                torchvision.transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225]),
+            ]
+        )
+
+        # preprocess and prepare image tensor
+        img = tfms(img).unsqueeze(0).to(device)
+
+        # initialize a pretrained model
+        net = test_pretrained_networks(EfficientNetBN, input_param, device)
+
+        # run inference
+        with eval_mode(net):
+            result = net(img)
+        pred_label = torch.argmax(result, dim=-1)
+
+        # check output
+        self.assertEqual(pred_label, expected_label)
+
+    def test_drop_connect_layer(self):
+        p_list = [float(d + 1) / 10 for d in range(9)]
+        # testing 1D, 2D and 3D shape
+        for rand_tensor_shape in [(512, 16, 4), (384, 16, 4, 4), (256, 16, 4, 4, 4)]:
+
+            # test validation mode, out tensor == in tensor
+            training = False
+            for p in p_list:
+                in_tensor = torch.rand(rand_tensor_shape) + 0.1
+                out_tensor = drop_connect(in_tensor, p, training=training)
+                self.assertTrue(torch.equal(out_tensor, in_tensor))
+
+            # test training mode, sum(out tensor != in tensor)/out_tensor.size() == p
+            # use tolerance of 0.175 to account for rounding errors due to finite set in/out
+            tol = 0.175
+            training = True
+            for p in p_list:
+                in_tensor = torch.rand(rand_tensor_shape) + 0.1
+                out_tensor = drop_connect(in_tensor, p, training=training)
+
+                p_calculated = 1.0 - torch.sum(torch.isclose(in_tensor, out_tensor * (1.0 - p))) / float(in_tensor.numel())
+                p_calculated = p_calculated.cpu().numpy()
+
+                self.assertTrue(abs(p_calculated - p) < tol)
+
+    def test_ill_arg(self):
+        with self.assertRaises(ValueError):
+            # wrong spatial_dims
+            EfficientNetBN(model_name="efficientnet-b0", spatial_dims=4)
+            # wrong model_name
+            EfficientNetBN(model_name="efficientnet-b10", spatial_dims=3)
+
+    def test_func_get_efficientnet_input_shape(self):
+        for model in get_model_names():
+            result_shape = get_efficientnet_image_size(model_name=model)
+            expected_shape = get_expected_model_shape(model)
+            self.assertEqual(result_shape, expected_shape)
+
+    def test_script(self):
+        net = EfficientNetBN(model_name="efficientnet-b0", spatial_dims=2, in_channels=3, num_classes=1000)
```

Comment:
@yiheng-wang-nv this one mostly supports torchscript, are we good here?

Reply:
Yes, I think this case is good!